doc/source/whatsnew/v1.3.0.rst

```diff
@@ -271,6 +271,7 @@ I/O
 - Allow custom error values for parse_dates argument of :func:`read_sql`, :func:`read_sql_query` and :func:`read_sql_table` (:issue:`35185`)
 - Bug in :func:`to_hdf` raising ``KeyError`` when trying to apply
   for subclasses of ``DataFrame`` or ``Series`` (:issue:`33748`).
+- Bug in :func:`put` raising a wrong ``TypeError`` when saving a DataFrame with non-string dtype (:issue:`34274`)
```

Comment:
this needs to be ``` :meth:`~HDFStore.put` ```

say this is an improved error message

Reply:
Oh thank you for that
It's morning in my timezone here and I just wake up 😅

Reply:
heheh, no prob this PR looks good. just a small change.